crates/biome_js_analyze/src/lint/correctness/no_undeclared_dependencies.rs

```diff
@@ -34,32 +40,186 @@ declare_lint_rule! {
     /// ```js,ignore
     /// import assert from "node:assert";
     /// ```
+    ///
+    /// ## Options
+    ///
+    /// This rule supports the following options:
+    /// - `devDependencies`: If set to `false`, then the rule will show an error when `devDependencies` are imported. Defaults to `true`.
+    /// - `peerDependencies`: If set to `false`, then the rule will show an error when `peerDependencies` are imported. Defaults to `true`.
+    /// - `optionalDependencies`: If set to `false`, then the rule will show an error when `optionalDependencies` are imported. Defaults to `true`.
+    ///
+    /// You can set the options like this:
+    /// ```json
+    /// {
+    ///   "options": {
+    ///     "devDependencies": false,
+    ///     "peerDependencies": false,
+    ///     "optionalDependencies": false
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// You can also use an array of globs instead of literal booleans:
+    /// ```json
+    /// {
+    ///   "options": {
+    ///     "devDependencies": ["**/*.test.js", "**/*.spec.js"]
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// When using an array of globs, the setting will be set to `true` (no errors reported)
+    /// if the name of the file being linted (i.e. not the imported file/module) matches a single glob
+    /// in the array, and `false` otherwise.
```

Comment:
```suggestion
    /// You can also use an array of globs instead of literal booleans.
    /// When using an array of globs, the setting will be set to `true` (no errors reported)
    /// if the name of the file being checked (i.e. not the imported file/module) matches a single glob
    /// in the array, and `false` otherwise.
    ///
    /// TODO: explain the example here
    ///
    /// ```json
    /// {
    ///   "options": {
    ///     "devDependencies": ["tests/*.test.js", "tests/*.spec.js"]
    ///   }
    /// }
    /// ```
    ///

```

Reply:
I left a `TODO` at line 67. Usually, in our docs, we follow this pattern:
- explain the options
- explain the example
- show code of the example

The `TODO` should explain the example of the code. Also, I removed `**/` because it usually matches too much and replaced it with a more simple glob